Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -270,6 +270,15 @@ public enum GeneralTraitsOptions {
     IsHidden = true
   };
 
+  public enum InputTypeEnum {
+    Source,
+    Binary
+  }
+
+  public static readonly Option<InputTypeEnum> InputType = new("--input-format", () => InputTypeEnum.Source) {
```

Comment:
Do we need this? I think it would be simpler to just determine this from the file extension, as we already do for `.dfy` vs `.doo`, and even `.toml` for project files in some places.

(We may want to bikeshed whether `.dbin` is the best extension name, but that's a separate discussion :)

Reply:
I'm using it in combination with `--stdin`. For the Java-Dafny compiler, so far I decided not to create any intermediate files but stream directly to `dafny`.